src/connector/Cargo.toml

```
@@ -23,6 +23,7 @@ arrow-schema = { workspace = true }
 arrow-schema-iceberg = { workspace = true }
 arrow-select = { workspace = true }
 assert_matches = "1"
+async-compression = "0.4.5"
```

Seems that the "tokio" feature gate should be enabled.

```suggestion
async-compression = { version = "0.4.5", features = ["tokio"] }
```